src/algebra/group/type_tags.lean

```diff
@@ -28,17 +28,33 @@ def additive (α : Type*) := α
 multiplicative structure. -/
 def multiplicative (α : Type*) := α
 
+namespace additive
+
 /-- Reinterpret `x : α` as an element of `additive α`. -/
-def additive.of_mul : α ≃ additive α := ⟨λ x, x, λ x, x, λ x, rfl, λ x, rfl⟩
+def of_mul : α ≃ additive α := ⟨λ x, x, λ x, x, λ x, rfl, λ x, rfl⟩
 
 /-- Reinterpret `x : additive α` as an element of `α`. -/
-def additive.to_mul : additive α ≃ α := additive.of_mul.symm
+def to_mul : additive α ≃ α := of_mul.symm
+
+@[simp] lemma of_mul_symm_eq : (@of_mul α).symm = to_mul := rfl
```

Comment:
But this way, if you end up with a `to_mul (of_mul x)`, it doesn't simp away to `x`. Could you please add simp-lemmas for those combis?

Reply:
Those lemmas already exist, don't they? `of_add_to_add` and vice versa.

Reply:
Ooh sorry, I didn't realise that. My bad.

bors merge